common/protob/messages-cardano.proto

```diff
@@ -112,149 +121,208 @@ message CardanoPublicKey {
-        required CardanoPoolRelayType type = 1;   // pool relay type
-        optional bytes ipv4_address = 2;          // ipv4 address of the relay given as 4 bytes
-        optional bytes ipv6_address = 3;          // ipv6 address of the relay given as 16 bytes
-        optional string host_name = 4;            // relay host name given as URL, at most 64 characters
-        optional uint32 port = 5;                 // relay port number in the range 0-65535
-    }
-
-    /**
-     * Stake pool metadata parameters
-     */
-    message CardanoPoolMetadataType {
-        required string url = 1;   // stake pool url hosting metadata, at most 64 characters
-        required bytes hash = 2;   // stake pool metadata hash
-    }
-
-    /**
-     * Stake pool parameters
-     */
-    message CardanoPoolParametersType {
-        required bytes pool_id = 1;                           // stake pool cold public key hash (28 bytes)
-        required bytes vrf_key_hash = 2;                      // VRF key hash (32 bytes)
-        required uint64 pledge = 3;                           // pledge amount in lovelace
-        required uint64 cost = 4;                             // cost in lovelace
-        required uint64 margin_numerator = 5;                 // pool margin numerator
-        required uint64 margin_denominator = 6;               // pool margin denominator
-        required string reward_account = 7;                   // bech32 reward address where the pool receives rewards
-        repeated CardanoPoolOwnerType owners = 8;             // pool owners list
-        repeated CardanoPoolRelayParametersType relays = 9;   // pool relays list
-        optional CardanoPoolMetadataType metadata = 10;       // pool metadata
-    }
-
-    /**
-     * Structure representing cardano transaction certificate
-     */
-     message CardanoTxCertificateType {
-        required CardanoCertificateType type = 1;                 // certificate type
-        repeated uint32 path = 2;                                 // BIP-32 path to derive (staking) key
-        optional bytes pool = 3;                                  // pool hash
-        optional CardanoPoolParametersType pool_parameters = 4;   // used for stake pool registration certificate
-    }
-    /**
-     * Structure representing cardano transaction withdrawals
-     */
-    message CardanoTxWithdrawalType {
-        repeated uint32 path = 1;
-        required uint64 amount = 2;
-    }
-
-    message CardanoCatalystRegistrationParametersType {
-        required bytes voting_public_key = 1;
-        repeated uint32 staking_path = 2;
-        required CardanoAddressParametersType reward_address_parameters = 3;
-        required uint64 nonce = 4;
-    }
-
-    message CardanoTxAuxiliaryDataType {
-        optional bytes blob = 1;
-        optional CardanoCatalystRegistrationParametersType catalyst_registration_parameters = 2;
-    }
+    required CardanoTxSigningMode signing_mode = 1;
```

Comment:
if the meaning of the fields changes, Trezor devs prefer to comment out the old ones and introduce new ones with a non-colliding id - it prevent weird behavior with older versions of Trezor that would try to interpret the same key differently/run into low-level wire-incompatibility issues and also that way track of changes to the format is kept.

Reply:
I've seen this and even Google recommends not reusing the field ids in the same message.
But I've also read that the [encoding of field ids over 15 is less efficient](https://developers.google.com/protocol-buffers/docs/overview#assigning_field_numbers) and I'd comment out most of the existing messages, so we thought it's ok to make `CardanoSignTx` a new message (with the same name, but a new message id), so we can reassign the field ids. 
https://github.com/vacuumlabs/trezor-firmware/blob/01c3cbfd8dff29eaef8417d834b283cc8ee2e0b1/common/protob/messages.proto#L215
Do you think this is ok, or should or should I stick to commenting out the fields?

Reply:
how would this work from trezor-connect perspective? I guess we need to support both the old version and the new one of the CardanoSignTx message at some point in trezor-connect to allow for a continuous transition of users from the old firmware to the new one, otherwise the integrating parties would need to switch from old connect to the new connect right when the new trezor firmware is rolled out (which may even be gradual, so some some users would be unable unable to work with their trezor devices at all)

So perhaps I'd consider keeping the original CardanoSignTx message (and tag it as deprecated - found this in BTC: https://github.com/trezor/trezor-firmware/blob/f9d4be268ead711c8324b471a8911cdd9a3a8c19/common/protob/messages-bitcoin.proto#L256) and the new one can be called CardanoSignTxInit (as the initial message for the cardanoSignTx call).

Reply:
I see. And in this case we actually have to keep & deprecate all the old messages (`CardanoSignedTxChunk`, `CardanoSignedTx`, etc.), not just the `CardanoSignTx`, right?